services/wfm/src/test/java/org/openkilda/wfm/isl/DiscoveryManagerTest.java

```diff
@@ -46,103 +59,209 @@
+        dstNode1 = new NetworkEndpoint("sw3", 1);
+        srcNode2 = new NetworkEndpoint("sw1", 2);
+        dstNode2 = new NetworkEndpoint("sw3", 2);
+        srcNode3 = new NetworkEndpoint("sw2", 1);
+        dstNode3 = new NetworkEndpoint("sw3", 3);
+
+        dm.handlePortUp(srcNode1.getDatapath(), srcNode1.getPortNumber());
+        dm.handlePortUp(srcNode2.getDatapath(), srcNode2.getPortNumber());
+        dm.handlePortUp(srcNode3.getDatapath(), srcNode3.getPortNumber());
     }
 
-
     @Test
-    public void makeDiscoveryPlan() {
-        // The discovery plan should test every link that isn't forlorned and where sufficient ticks
-        // have passed. Additionally, the disco plan determines what is part of the failure notification
-        // plan.
-        //
-        // Tests:
-        //  1) all ports that should send a disco packet sent are in the needDiscovery list.
-        //  2) all ports that haven't had a response within timeout are in the discoveryFailure list.
-        //
-        List<DiscoveryLink> nodes;
+    public void shouldDiscoveryPlanContainsAllEndpoints() {
         setupThreeLinks();
 
         // Initially, given 0 tick interval, everything should be in discoveryPlan and no failures
         DiscoveryManager.Plan discoveryPlan = dm.makeDiscoveryPlan();
         assertEquals(3, discoveryPlan.needDiscovery.size());
         assertEquals(0, discoveryPlan.discoveryFailure.size());
+    }
 
-        // In order for one of them to be in failure plan, it has to have been a success beforehand
-        dm.handleDiscovered(srcNode1.getSwitchDpId(), srcNode1.getPortId(),
-                dstNode1.getSwitchDpId(), dstNode1.getPortId()); // clear attempts
-        dm.handleDiscovered(srcNode2.getSwitchDpId(), srcNode2.getPortId(),
-                dstNode2.getSwitchDpId(), dstNode2.getPortId()); // clear attempts
-        dm.handleDiscovered(srcNode3.getSwitchDpId(), srcNode3.getPortId(),
-                dstNode3.getSwitchDpId(), dstNode3.getPortId()); // clear attempts
-        dm.makeDiscoveryPlan();                                      // increase attempts of all to 1
-        dm.handleDiscovered(srcNode1.getSwitchDpId(), srcNode1.getPortId(),
-                dstNode1.getSwitchDpId(), dstNode1.getPortId()); // clear attempts
-        dm.handleDiscovered(srcNode2.getSwitchDpId(), srcNode2.getPortId(),
-                dstNode2.getSwitchDpId(), dstNode2.getPortId()); // clear attempts
-        dm.makeDiscoveryPlan();                                      // attempts: link1&2 @ 1, link3 @ 2
-        discoveryPlan = dm.makeDiscoveryPlan();                      // the attempts test is based on the 1 & 2 count
-        assertEquals(3, discoveryPlan.needDiscovery.size());
+    @Test
+    public void shouldAlwaysSendDiscoveryForActiveLinks() {
+        setupThreeLinks();
+
+        for (int attempt = 1; attempt <= maxAttemptsLimit + islHealthFailureLimit + 1; attempt++) {
+            if (attempt == 1) {
```

Comment:
You can `verifyAllLinks(false);` before the loop starts and remove this whole if-else clause from here, only do `verifyAllLinks(true);` on line 131